packages/block-editor/src/components/inner-blocks/use-block-context.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+import { store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * Returns a context object for a given block.
+ *
+ * @param {string} clientId The block client ID.
+ *
+ * @return {Record<string,*>} Context value.
+ */
+export default function useBlockContext( clientId ) {
+	return useSelect(
+		( select ) => {
+			const block = select( blockEditorStore ).getBlock( clientId );
+			if ( ! block ) {
+				return undefined;
+			}
+
+			const blockType = select( blocksStore ).getBlockType( block.name );
+			if ( ! blockType ) {
+				return undefined;
+			}
+
+			if ( Object.keys( blockType.providesContext ).length === 0 ) {
```

Comment:
Any chance `provideContext` will not exist, causing `Object.keys( undefined )` to crash?

Reply:
When `provideContext` doesn't exist in `block.json`, it will get defaulted to an empty object [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/src/api/registration.js#L291).

There only one edge case: if I specify `provideContext: null` in `block.json`, it will not get defaulted and `Object.keys` will really crash.

Reply:
Maybe it's worth adding a `?? {}` or an additional non-nullish check just in case.

Reply:
I realized there's the `providesContext: null` case only after your comment.

Reply:
That sounds like intentionally breaking a block. What if someone sets a string instead of an object?

I think this is something that block register should handle.

Reply:
> I think this is something that block register should handle.

I don't disagree, after all, we probably have many places where we expect a string value in `block.json`, and do unguarded `.includes` or `.startsWith` calls. A malicious `null` will crash them, too.

Reply:
Agreed. That's why I approved the PR in the first place and considered this a minor thing. Leaving it to your judgement @Mamaduka.

Reply:
Thanks again for the great feedback. I will merge this as it is. But happy to add extra checks in the future. Maybe as a part of block registration.